src/pcrlock/pcrlock.c

```diff
@@ -5046,9 +5070,9 @@ static int parse_argv(int argc, char *argv[]) {
                 }
 
                 case ARG_RECOVERY_PIN:
-                        r = parse_boolean_argument("--recovery-pin", optarg, &arg_recovery_pin);
-                        if (r < 0)
-                                return r;
+                        arg_recovery_pin = recovery_pin_mode_from_string(optarg);
```

Comment:
Isn't this losing compat with the previous boolean behaviour?

Reply:
no, it's fully compatible, we still parse booleans, and map them to the same behaviour.

(the tool is marked as experimental in the NEWS file btw, hence even if it would change interface – which it actually does not – it should be OK)

Reply:
ok, it wasn't clear to me looking at the impl here